src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -1392,8 +1401,36 @@ def nunique(self) -> None:
         --------
         DataFrame.nunique: Method nunique for DataFrame.
         Series.count: Count non-NA/null observations in the Series.
-        """
-        # TODO: SNOW-1458132 implement nunique
+
+        Examples
+        --------
+        >>> s = pd.Series([1, 3, 5, 7, 7])
+        >>> s
+        0    1
+        1    3
+        2    5
+        3    7
+        4    7
+        dtype: int64
+
+        >>> s.nunique()
+        4
+        """
+        return (
+            SnowflakeQueryCompiler(
+                self._query_compiler._modin_frame.append_column(
```

Comment:
at frontend layer, we are only suppose to access the query_compiler layer not the  _modin_frame layer. it seems we are just trying to get a data column that is the same as index column, then reuse the nunique, it is almost similar as doing reset_index, or insert a column to the index dataframe.

However, is the purpose to try avoid extend the nunique interface at query compiler, i don't think it worth it, nobody says the nuique implementaiton at query backend can only be applied on data columns. Or you can just add another  index_nunique function at query compiler, that reuses some funcitonality of the query compiler

Reply:
I think the reason it's done here is to try to have as few changes as possible between upstream modin and this code base. I'll make this a new query compiler function.

Reply:
Is this better?

Reply:
I am not i don't think that change is related to upstream modin, since this is completely new to modin. if you are talking about compiler interface, i can accept this argument, but then the way to resolve this is to have two different compiler interface there

Reply:
When we upstream this change to Modin, I recommend adding a method to add the index column as a data column, and not extending the interface of `unique`. That way, we can minimize the query compiler changes while also not accessing the private `_modin_frame`.

Generally we would like to avoid extending the query compiler interface with features that are only useful for indexes.